listenbrainz_spark/hdfs/__init__.py

```diff
@@ -57,31 +59,52 @@ def upload_archive(self, tmp_dump_dir, tar, dest_path, schema, callback=None, fo
                 force: If True deletes dir at dest_path
         """
         if callback is None:
-            raise NotImplementedError('Callback to process JSON missing. Aboritng...')
-
-        if force:
-            current_app.logger.info('Removing {} from HDFS...'.format(dest_path))
-            utils.delete_dir(dest_path, recursive=True)
-            current_app.logger.info('Done!')
+            raise NotImplementedError('Callback to process JSON missing. Aborting...')
 
-        file_count = 0
+        current_app.logger.info("Uploading listens to temporary directory in HDFS...")
+        total_files = 0
         total_time = 0.0
         for member in tar:
             if member.isfile() and self._is_json_file(member.name):
-                current_app.logger.info('Loading {}...'.format(member.name))
-                t = time.time()
-                tar.extract(member)
+                current_app.logger.info("Uploading {}...".format(member.name))
+                t0 = time.time()
+
+                try:
+                    tar.extract(member)
+                except TarError as err:
+                    current_app.logger.error("{} while extracting {}, aborting import".format(
+                        type(err).__name__, member.name), exc_info=True)
+                    # Cleanup
+                    if utils.path_exists('/temp'):
+                        utils.delete_dir('/temp', recursive=True)
+                    if utils.path_exists(tmp_dump_dir):
+                        utils.delete_dir(tmp_dump_dir, recursive=True)
+                    shutil.rmtree(tmp_dump_dir)
+                    return
```

Comment:
If we just return here, the handler on the spark_reader side will not know that the import failed, we should raise an exception here, catch it in the handler and send back an appropriate message.

Reply:
We do not send back anything to the `spark_reader` but log the error on sentry for any other request. So I have raised `DumpInvalidException` with appropriate message so that it gets logged in sentry instead of returning.